test/unittests/evm_calls_test.cpp

```diff
@@ -293,6 +296,72 @@ TEST_F(evm, call_new_account_create)
     EXPECT_EQ(call_msg.gas, 6000);
 }
 
+TEST_F(evm, call_new_account_creation_cost)
+{
+    const auto call_dst = evmc::address{{0xad}};
+    const auto code = 4 * push(0) + calldataload(0) + push({call_dst.bytes, sizeof(call_dst)}) +
+                      push(0) + OP_CALL + ret_top();
+    msg.destination = evmc_address{{3}};
+
+
+    rev = EVMC_TANGERINE_WHISTLE;
+    accounts[msg.destination].set_balance(0);
+    execute(code, "00");
+    EXPECT_GAS_USED(EVMC_SUCCESS, 25000 + 739);
+    EXPECT_OUTPUT_INT(1);
+    ASSERT_EQ(recorded_calls.size(), 1);
+    EXPECT_EQ(recorded_calls.back().destination, call_dst);
+    EXPECT_EQ(recorded_calls.back().gas, 0);
+    ASSERT_EQ(recorded_account_accesses.size(), 2);
+    EXPECT_EQ(recorded_account_accesses[0], call_dst);  // Account exist?
+    EXPECT_EQ(recorded_account_accesses[1], call_dst);  // Call.
+    recorded_account_accesses.clear();
+    recorded_calls.clear();
+
+    rev = EVMC_TANGERINE_WHISTLE;
+    accounts[msg.destination].set_balance(1);
+    execute(code, "0000000000000000000000000000000000000000000000000000000000000001");
+    EXPECT_GAS_USED(EVMC_SUCCESS, 25000 + 9000 + 739);
+    EXPECT_OUTPUT_INT(1);
+    ASSERT_EQ(recorded_calls.size(), 1);
+    EXPECT_EQ(recorded_calls.back().destination, call_dst);
+    EXPECT_EQ(recorded_calls.back().gas, 2300);
+    ASSERT_EQ(recorded_account_accesses.size(), 3);
+    EXPECT_EQ(recorded_account_accesses[0], call_dst);         // Account exist?
+    EXPECT_EQ(recorded_account_accesses[1], msg.destination);  // Balance.
+    EXPECT_EQ(recorded_account_accesses[2], call_dst);         // Call.
+    recorded_account_accesses.clear();
+    recorded_calls.clear();
+
+    rev = EVMC_SPURIOUS_DRAGON;
+    accounts[msg.destination].set_balance(0);
+    execute(code, "00");
+    EXPECT_GAS_USED(EVMC_SUCCESS, 739);
+    EXPECT_OUTPUT_INT(1);
```

Comment:
This is the value left on the stack by the CALL (the `ret_top()` macro returns the top stack item). The CALL puts 1 on the stack on success, 0 on failure. We expect it to succeed in this case because this is set as that in `.call_result` (the default value used here).